### PR TITLE
Fix: vtex search params IS/Legacy

### PR DESCRIPTION
**Pull request by IncognitaDev**
###  Search with path / Empty search in vtex IS

When a url path does not contain `s` query and its not a page/category/department the return was a seach page with all products. To fix, I add the default vtex behavior, any path that is not a page its coverted by a search query, so when its a not valid query the return is a not found (if implemented by storefront) and when is, the return is the search return.

### Search with path in vtex Legacy

As said above, a path thats is no a page its a valid search term, the behavior is the same.

### CODE REVIEW

File: vtex/loaders/intelligentSearch/productListingPage.ts [hunk 1]
```diff
@@ -250,18 +250,41 @@ const loader = async (
...
+  /**
+   *  verify if when url its not a category/department/collection
+   *  and is not a default query format but is a valid search based in default IS behavior on native stores
+   */
+  if (
+    !pageTypes.length &&
+    !selected.length &&
+    !args.query &&
+    pathList.length == 1
+  ) {
```
Comment (lucis) [inline vtex/loaders/intelligentSearch/productListingPage.ts:278]: Create a variable with this condition and name it appropriately

File: vtex/loaders/intelligentSearch/productListingPage.ts [hunk 2]
```diff
@@ -250,18 +250,41 @@ const loader = async (
...
+    args.query = pathList[0];
```
Comment (lucis) [inline vtex/loaders/intelligentSearch/productListingPage.ts:279]: Can you avoid mutating the object and create a new const?

Comment (IncognitaDev) [reply vtex/loaders/intelligentSearch/productListingPage.ts:276]: I also think that mutating its not good, I just cant think in a name for new args const

File: vtex/loaders/intelligentSearch/productListingPage.ts (hunk 1 above)
Comment (IncognitaDev) [reply vtex/loaders/intelligentSearch/productListingPage.ts]: done.

File: vtex/loaders/intelligentSearch/productListingPage.ts (hunk 2 above)
Comment (IncognitaDev) [reply vtex/loaders/intelligentSearch/productListingPage.ts]: done

**Review by lucis — APPROVED** [review]

**Pull request merged**